src/main/java/net/dv8tion/jda/entities/impl/GuildImpl.java

```diff
@@ -325,6 +325,26 @@ public TextChannel getPublicChannel()
     }
 
     @Override
+    public TextChannel getTextChannelById(String id)
+    {
+        return getJDA().getTextChannelById(id) == null 
+            ? null 
+            : getTextChannels().contains(getJDA().getTextChannelById(id)) 
+                ? getJDA().getTextChannelById(id) 
+                : null;
```

Comment:
I am also confused as to why you do this. Within GuildImpl exists `textChannels` and `voiceChannels`, both of which are HashMaps which you can use a channel's id to do lookups.
These methods should not event touch the JDA object. Please refactor.